### DISCUSSION TITLE
OL and UL tag support with correct indentation

### DISCUSSION THREAD

**Discussion by ali-thowfeek** [ali-thowfeek/simple_html_css_flutter#12] [Ideas]
@nohli This is how I first started to experiment. f2afcc2
But there are some problems to be dealt with. Need to test it thoroughly.
For instance, If there is a tag in-between the `<li>` tag like, `<li>First <b>element</b></li>` it breaks it into a new line.

**Comment by nohli** [comment]
First try handling li tags using your code :)

https://github.com/ali-thowfeek/simple_html_css_flutter/tree/handle-li-tags

**Reply by ali-thowfeek** [reply]
My initial method didn't work, because we were trying to handle bullets within this `for (final XmlEvent event in _events)` loop. So now, I moved the handling out of there and it was able to resolve this issue.
> For instance, If there is a tag in-between the `<li>` tag like, `<li>First <b>element</b></li>` it breaks it into a new line.

So far it is promising. But, I have only written code to handle the first `li` item. We'll have to figure out a way to apply it to all.
(Also, sorry for the messy code😉, we'll clean it out once we get it right)

**Reply by nohli** [reply]
The latest commit includes all li tags :)

**Reply by nohli** [reply]
Right now, above a list there are always two empty lines. Should only be one. Maybe ul gets one and li also gets one new line.

**Reply by ali-thowfeek** [reply]
Well, the starting tags don't get a new line. They are the new lines of a tag above like p, or div. 
Although, I think it is ok being that way. Look at the following example.
https://www.w3schools.com/code/tryit.asp?filename=GOQQ1EAIF83I

**Reply by ali-thowfeek** [reply]
And on a side note the package also renders any text or white space in the html code between the tags. For example if you copy some html from somewhere with **indentation** and put it in our string then you would be able to see how it misbehaves. I don't know if we should let it be or remove only the whitespaces.

**Comment by ali-thowfeek** [comment]
Im working on the font size of the bullets right now.

**Reply by ali-thowfeek** [reply]
I've done it 62930ef
For now, i've taken font size of the **first** span of the list item, and based on that the font size, the baseline and the padding of the bullet gets determined. But there could be a problem if theres a `<span>` or `<a>` or any other **tag in the start** with a bigger or smaller font size to the others, like in the following example:
```
<li><a style="font-size:7px">Lorem</a> ipsum dolor sit amet.</li>
```
in the above eg: the bullet and the anchor tag would be rendered smaller and the rest would be bigger, which might look odd.

We'll have to think through how we want it to be. Maybe this might not occur as frequently an we could just ignore it or maybe we'll have to make it more efficient .
@nohli any thoughts
